src/SablierV2MerkleLockupFactory.sol

```diff
@@ -62,13 +164,25 @@ contract SablierV2MerkleLockupFactory is ISablierV2MerkleLockupFactory {
             }
         }
 
-        // Check: the sum of percentages equals 100%.
-        if (totalPercentage != uUNIT) {
-            revert Errors.SablierV2MerkleLockupFactory_TotalPercentageNotOneHundred(totalPercentage);
-        }
+        // Hash the parameters to generate a salt.
+        bytes32 salt = keccak256(
+            abi.encodePacked(
+                baseParams.asset,
+                baseParams.cancelable,
+                baseParams.expiration,
+                baseParams.initialAdmin,
+                abi.encode(baseParams.ipfsCID),
+                baseParams.merkleRoot,
+                bytes32(abi.encodePacked(baseParams.name)),
+                baseParams.transferable,
+                lockupTranched,
+                totalPercentage
```

Comment:
There's no value in including the total percentage in the salt. In the vast majority of cases, it will be 100%, so it won't add any differentiation.

Reply:
A corollary of not needing the percentage for the salt is that we can move the calculations in the MerkleLT contract's constructor.

Reply:
It would create two `for` loops in `createMerkleLT` execution. Also if we do not include `totalPercentage` in `salt`, should we include `tranchesWithPercentages`? All parameters being equal, two airstreams with different `tranchesWithPercentages` would lead to collision.

Reply:
nvm. There were already two `for` loops.

Reply:
but thoughts on adding `tranchesWithPercentages` in `salt`?

Reply:
since we include `streamDurations` in LL salt, we can also include `tranchesWithPercentages` in LT salt

Reply:
Since there is no objection: https://github.com/sablier-labs/v2-periphery/pull/355/commits/321a6da017d6ba39964859cca44dcc3f862e57fa

Reply:
happy to include `tranchesWithPercentages` in the salt